From 4729605f30a924990eb49a373b6782cd7e9ad6cd Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Tue, 12 Feb 2019 19:00:28 +0100
Subject: [v8 7.6.303.22] Wrap several native binary calls on Yocto building
 with qemu.

The patch below makes the V8 binaries run during the build be invoked through
QEMU, as they are built for the target:
* mksnapshot
* torque
* gen-regexp-special-case

Based on original patch from Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>

Upstream-Status: Inappropriate [embedded specific]

diff --git a/BUILD.gn b/BUILD.gn
index 8640517ae5..9d75d47699 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1075,6 +1075,7 @@ action("run_torque") {
   }
 
   args = [
+    "./v8-qemu-wrapper.sh",
     "./" + rebase_path(get_label_info(":torque($v8_generator_toolchain)",
                                       "root_out_dir") + "/torque",
                        root_build_dir),
@@ -1153,6 +1154,7 @@ action("generate_bytecode_builtins_list") {
     ":bytecode_builtins_list_generator($v8_generator_toolchain)",
   ]
   args = [
+    "./v8-qemu-wrapper.sh",
     "./" + rebase_path(
             get_label_info(
                     ":bytecode_builtins_list_generator($v8_generator_toolchain)",
@@ -1196,6 +1198,7 @@ template("run_mksnapshot") {
     data = []
 
     args = [
+      "./v8-qemu-wrapper.sh",
       "./" + rebase_path(get_label_info(":mksnapshot($v8_snapshot_toolchain)",
                                         "root_out_dir") + "/mksnapshot",
                          root_build_dir),
@@ -3928,6 +3931,7 @@ action("run_gen-regexp-special-case") {
   ]
 
   args = [
+    "./v8-qemu-wrapper.sh",
     "./" + rebase_path(
             get_label_info(":gen-regexp-special-case($v8_generator_toolchain)",
                            "root_out_dir") + "/gen-regexp-special-case",
-- 
2.20.1

